File: 1.Array/BestTimeToBuyAndSellStock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 * https://leetcode.com/problems/best-time-to-buy-and-sell-stock/submissions/1668971580/
 */
var maxProfit = function(prices) {
    let ans = 0;
    let min_price = prices[0];

    for(let i = 0; i< prices.length; i++){
        let profit = prices[i] - min_price;
        ans = Math.max(ans, profit);
        min_price = Math.min(min_price, prices[i]);
    }
    return ans;
};

File: 1.Array/BestTimeToBuyAndSellStock/solution.java
/**
 * https://leetcode.com/problems/best-time-to-buy-and-sell-stock/submissions/1668986235/
 */
class Solution {
    public int maxProfit(int[] prices) {
       int ans = 0;
       int minPrice = prices[0];

       for(int i = 0; i < prices.length; i++){
        int profit = prices[i] - minPrice;
        ans = Math.max(ans, profit);
        minPrice = Math.min(minPrice, prices[i]);
       }
       return ans;
    }
}